src/cpp/rtps/builtin/discovery/endpoint/EDPStatic.cpp

```diff
@@ -214,9 +214,35 @@ bool EDPStatic::removeLocalWriter(
 void EDPStatic::assignRemoteEndpoints(
         const ParticipantProxyData& pdata)
 {
+    GUID_t persistence_guid;
+
+    // Fill persistence GUID if present in UserData.
+    bool is_persistent_guid = (18 <= pdata.m_userData.size()) &&
+            ('V' == pdata.m_userData.at(0)) && ('G' == pdata.m_userData.at(1)) && ('W' == pdata.m_userData.at(2));
+    if (is_persistent_guid)
+    {
+        persistence_guid.guidPrefix.value[0] = pdata.m_userData.at(3);
+        persistence_guid.guidPrefix.value[1] = pdata.m_userData.at(4);
+        persistence_guid.guidPrefix.value[2] = pdata.m_userData.at(5);
+        persistence_guid.guidPrefix.value[3] = pdata.m_userData.at(6);
+        persistence_guid.guidPrefix.value[4] = pdata.m_userData.at(7);
+        persistence_guid.guidPrefix.value[5] = pdata.m_userData.at(8);
+        persistence_guid.guidPrefix.value[6] = pdata.m_userData.at(9);
+        persistence_guid.guidPrefix.value[7] = pdata.m_userData.at(10);
+        persistence_guid.guidPrefix.value[8] = pdata.m_userData.at(11);
+        persistence_guid.guidPrefix.value[9] = pdata.m_userData.at(12);
+        persistence_guid.guidPrefix.value[10] = pdata.m_userData.at(13);
+        persistence_guid.guidPrefix.value[11] = pdata.m_userData.at(14);
+        persistence_guid.entityId.value[0] = pdata.m_userData.at(15);
+        persistence_guid.entityId.value[1] = pdata.m_userData.at(16);
+        persistence_guid.entityId.value[2] = pdata.m_userData.at(17);
+    }
+
     for (ParameterPropertyList_t::const_iterator pit = pdata.m_properties.begin();
             pit != pdata.m_properties.end(); ++pit)
     {
+        persistence_guid.entityId.value[3] = 0;
```

Comment:
value[3] is the `entityKind`, so I think the user id should be assigned to either index 0 or 2

Reply:
This is actually not an option because will break interoperability

Reply:
As talked offline, the persistence_guid is always set by the user, and doesn't need to follow the standard GUID rules.